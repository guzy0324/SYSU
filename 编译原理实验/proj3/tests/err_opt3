/* Optional test Case 3: duplicate "main" method */
program erropt3;
class myTest
{
	declarations
		int x, y;
	enddeclarations

	method void main()
	{
	return x;
	}
}

class test1
{
	method void main()
	{
	/* nothing */
	}
}
