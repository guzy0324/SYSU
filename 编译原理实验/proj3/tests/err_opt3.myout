Semantic Error--line: 22, symbol main: method and return value types are mismatched.
Semantic Error--line: 22, main() method already declared.
********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     myTest          0                            class                                                       
  2          x          1                         variable  type-tree                                            
  3          y          1                         variable  type-tree                                            
  4       main          1  proc-tree             procedure                                                      0
  5      test1          0                            class                                                       
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,5,"test1"]
  | +-[ClassDefOp]
  | | |       +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[IDNode,19,"main"]
  | | +-[BodyOp]
  | |   +-[DUMMYnode]
  +-[ClassOp]
    |   +-[STNode,1,"myTest"]
    | +-[ClassDefOp]
    | | |         +-[DUMMYnode]
    | | |       +-[ReturnOp]
    | | |       | | +-[DUMMYnode]
    | | |       | +-[VarOp]
    | | |       |   +-[STNode,2,"x"]
    | | |     +-[StmtOp]
    | | |     | +-[DUMMYnode]
    | | |   +-[BodyOp]
    | | |   | +-[DUMMYnode]
    | | | +-[MethodOp]
    | | | | |   +-[DUMMYnode]
    | | | | | +-[SpecOp]
    | | | | | | +-[DUMMYnode]
    | | | | +-[HeadOp]
    | | | |   +-[STNode,4,"main"]
    | | +-[BodyOp]
    | |   |       +-[DUMMYnode]
    | |   |     +-[CommaOp]
    | |   |     | | +-[DUMMYnode]
    | |   |     | +-[TypeIdOp]
    | |   |     |   +-[INTEGERTNode]
    | |   |   +-[CommaOp]
    | |   |   | +-[STNode,3,"y"]
    | |   | +-[DeclOp]
    | |   | | |     +-[DUMMYnode]
    | |   | | |   +-[CommaOp]
    | |   | | |   | | +-[DUMMYnode]
    | |   | | |   | +-[TypeIdOp]
    | |   | | |   |   +-[INTEGERTNode]
    | |   | | | +-[CommaOp]
    | |   | | | | +-[STNode,2,"x"]
    | |   | | +-[DeclOp]
    | |   | |   +-[DUMMYnode]
    | |   +-[BodyOp]
    | |     +-[DUMMYnode]
    +-[ClassOp]
      +-[DUMMYnode]
