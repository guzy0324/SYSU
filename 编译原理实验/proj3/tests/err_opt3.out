Semantic Error--line: 21, symbol main: redeclared.
********************************Symbol Table************************************

          Name Nest-Level  Tree-Node Predefined        Kind      Type      Value     Offset  Dimension     Argnum

  1     myTest          0                            class                                                       
  2          y          1                         variable  type-tree                                            
  3          x          1                         variable  type-tree                                            
  4       main          1  proc-tree             procedure                                                      0
  5      test1          0                            class                                                       
************* SYNTAX TREE PRINTOUT ***********

  +-[DUMMYnode]
R-[ProgramOp]
  |   +-[STNode,5,"test1"]
  | +-[ClassDefOp]
  | | |       +-[DUMMYnode]
  | | |     +-[StmtOp]
  | | |     | +-[DUMMYnode]
  | | |   +-[BodyOp]
  | | |   | +-[DUMMYnode]
  | | | +-[MethodOp]
  | | | | |   +-[DUMMYnode]
  | | | | | +-[SpecOp]
  | | | | | | +-[DUMMYnode]
  | | | | +-[HeadOp]
  | | | |   +-[IDNode,19,"main"]
  | | +-[BodyOp]
  | |   +-[DUMMYnode]
  +-[ClassOp]
    |   +-[STNode,1,"myTest"]
    | +-[ClassDefOp]
    | | |         +-[DUMMYnode]
    | | |       +-[ReturnOp]
    | | |       | | +-[DUMMYnode]
    | | |       | +-[VarOp]
    | | |       |   +-[STNode,3,"x"]
    | | |     +-[StmtOp]
    | | |     | +-[DUMMYnode]
    | | |   +-[BodyOp]
    | | |   | +-[DUMMYnode]
    | | | +-[MethodOp]
    | | | | |   +-[DUMMYnode]
    | | | | | +-[SpecOp]
    | | | | | | +-[DUMMYnode]
    | | | | +-[HeadOp]
    | | | |   +-[STNode,4,"main"]
    | | +-[BodyOp]
    | |   |       +-[DUMMYnode]
    | |   |     +-[CommaOp]
    | |   |     | | +-[DUMMYnode]
    | |   |     | +-[TypeIdOp]
    | |   |     |   +-[INTEGERTNode]
    | |   |   +-[CommaOp]
    | |   |   | +-[STNode,2,"y"]
    | |   | +-[DeclOp]
    | |   | | |     +-[DUMMYnode]
    | |   | | |   +-[CommaOp]
    | |   | | |   | | +-[DUMMYnode]
    | |   | | |   | +-[TypeIdOp]
    | |   | | |   |   +-[INTEGERTNode]
    | |   | | | +-[CommaOp]
    | |   | | | | +-[STNode,3,"x"]
    | |   | | +-[DeclOp]
    | |   | |   +-[DUMMYnode]
    | |   +-[BodyOp]
    | |     +-[DUMMYnode]
    +-[ClassOp]
      +-[DUMMYnode]
